stocks2.lua

```diff
@@ -0,0 +1,544 @@
+        if not filter_states.melt and item.flags.melt then goto continue end
+
+        local in_inv_char
+        if item.flags.in_inventory then
+            in_inv_char = 'I'
+            has_flags = true
+        else
+            in_inv_char = ' '
+        end
+        if not filter_states.in_inv and item.flags.in_inventory then goto continue end
+
+        -- skip if item has no flags and that filter is off!
+        if not filter_states.no_flags and has_flags == false then goto continue end
+
+        -- check quality
+        local quality_label, quality_pen
+        quality_label = quality_labels[item:getQuality() + 1]
+        quality_pen = quality_pens[item:getQuality() + 1]
+
+        -- skip by quality or wear
+        if item:getQuality() < min_qual or item:getQuality() > max_qual then goto continue end
+        if item:getWear() < min_wear then goto continue end
+
+        local wear_tags = {"", "x", "X", "XX"}
+
+        if not filter or #filter < 1 or string.match(string.lower(v.desc), filter) then
+            local full_item_desc
+            if item:getWear() > 0 then
+                local wear_tag = wear_tags[item:getWear() + 1]
+                full_item_desc = wear_tag .. v.desc .. wear_tag
+            else
+                full_item_desc = v.desc
+            end
+
+            table.insert(list_contents, {
+                text = {
+                    {text = full_item_desc, width = item_desc_width, rjustify = false, pad_char = ' '}, ' ',
+                    {text = job_char, width = 1, pen = COLOR_LIGHTBLUE},
+                    {text = rotten_char, width = 1, pen = COLOR_CYAN},
+                    {text = owned_char, width = 1, pen = COLOR_GREEN},
+                    {text = forbidden_char, width = 1, pen = COLOR_RED},
+                    {text = dump_char, width = 1, pen = COLOR_LIGHTMAGENTA},
+                    {text = on_fire_char, width = 1, pen = COLOR_LIGHTRED},
+                    {text = melt_char, width = 1, pen = COLOR_BLUE},
+                    {text = in_inv_char, width = 1, pen = COLOR_WHITE}, "   ",
+                    {text = quality_label, width = 13, rjustify = false, pad_char = ' ', pen = quality_pen},
+                },
+                ref = v.ref,
+            })
+        end
+
+        ::continue::
+    end
+
+    self.subviews.item_list:setChoices(list_contents)
+    self.subviews.item_count_label:setText(string.format("Items: %d", #list_contents))
+    self.subviews.actions_item_count_label:setText({
+            {text = "Actions (", pen = COLOR_BROWN}, {text = tostring(#list_contents), pen = COLOR_LIGHTGREEN}, {text = " Items)", pen = COLOR_BROWN}
+    })
+    dfhack.printerr("done.")
+end
+
+StocksScreen = defclass(StocksScreen, gui.ZScreen)
+StocksScreen.ATTRS({
+    focus_path = "stocks2"
+})
+
+function StocksScreen:init()
+    self:addviews({StocksWindow{}})
+end
+
+function StocksScreen:onDismiss()
+    view = nil
+end
+
+if dfhack_flags.module then
+    return
+end
+
```

Comment:
need to check for a loaded fort. Example: https://github.com/DFHack/scripts/blob/master/gui/blueprint.lua#L623